ncftp fails to build on nixos-unstable

**Describe the bug**
ncftp fails to build on nixos-unstable

**To Reproduce**
1. Try to install ncftp on current nixos-unstable

**Expected behavior**
ncftp build should succeed

**Screenshots**
```
Compiling ncftpbatch:                                                 [ERROR]
  gcc -D_REENTRANT -O2 -W -Wall -Wno-format-y2k -fstack-protector-all -D_FORTIF
  Y_SOURCE=2 -DLINUX=54059 -DLINUX_GLIBC=23100 -Dsh_util -DO_S="linux-x86_64-gl
  ibc2.31" -DSYSCONFDIR="/nix/store/aaakh5iza6n1b0p246gi6w5mkx6y561b-ncftp-3.2.
  6/etc" -DHAVE_CONFIG_H -DLINUX=54059 -DLINUX_GLIBC=23100 -I/build/ncftp-3.2.6
   -I../libncftp -I../Strn -I../sio -I/build/ncftp-3.2.6 -I/build/ncftp-3.2.6/l
  ibncftp -I/build/ncftp-3.2.6/sio -I/build/ncftp-3.2.6/Strn preffw.o util.o sp
  oolutil.o version.o gpshare.o bookmark.o ncftpbatch.c -o ../bin/ncftpbatch -L
  ../libncftp -L../Strn -L../sio -L/build/ncftp-3.2.6/libncftp -L/build/ncftp-3
  .2.6/sio -L/build/ncftp-3.2.6/Strn -lncftp -lStrn -lsio -lresolv
  impure path `/tmp/ccZlhqPW.o' used in link
  collect2: error: ld returned 1 exit status
```

**Additional context**

**Notify maintainers**
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

**Metadata**
```
❯ nix-shell -p nix-info --run "nix-info -m"                                                                                                                                                                                                                                    ~
these paths will be fetched (0.05 MiB download, 0.28 MiB unpacked):
  /nix/store/bgdqr36i8csfda3yfq5asbcnsdkwrklc-bash-interactive-4.4-p23-dev
copying path '/nix/store/bgdqr36i8csfda3yfq5asbcnsdkwrklc-bash-interactive-4.4-p23-dev' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.59, NixOS, 20.09.git.148e20e9844 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.7`
 - channels(root): `"nixos-20.09pre242769.61525137fd1"`
 - channels(kevin): `"nixos-20.09pre242769.61525137fd1"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/tools/networking/ncftp/default.nix
+++ b/pkgs/tools/networking/ncftp/default.nix
@@ -14,6 +14,8 @@ stdenv.mkDerivation rec {
   enableParallelBuilding = true;
 
   preConfigure = ''
+    find -name Makefile.in | xargs sed -i '/^TMPDIR=/d'
+
     find . -name '*.sh' -or -name '*.in' -or -name '*.c' -or -name configure | xargs sed -i \
       -e 's@/bin/ls@${coreutils}/bin/ls@g' \
       -e 's@/bin/rm@${coreutils}/bin/rm@g'

